File: usaco.old/temp/billboard.java

import java.io.*;
import java.util.Scanner;

public class billboard {
    public static void main(String[] args) throws IOException {
        int Ax1;
        int Ay1;
        int Ax2;
        int Ay2;
        int Bx1;
        int By1;
        int Bx2;
        int By2;
        int Cx1;
        int Cy1;
        int Cx2;
        int Cy2;
        Scanner scanner;
        PrintWriter out;
        scanner = new Scanner(new File("billboard.in"));
        Ax1 = scanner.nextInt();
        Ay1 = scanner.nextInt();
        Ax2 = scanner.nextInt();
        Ay2 = scanner.nextInt();
        Bx1 = scanner.nextInt();
        By1 = scanner.nextInt();
        Bx2 = scanner.nextInt();
        By2 = scanner.nextInt();
        Cx1 = scanner.nextInt();
        Cy1 = scanner.nextInt();
        Cx2 = scanner.nextInt();
        Cy2 = scanner.nextInt();
        System.out.printf("  :| Ax1=%s, Ay1=%s, Ax2=%s, Ay2=%s\n", Ax1, Ay1, Ax2, Ay2);
        int area;
        area = (Ax2 - Ax1) * (Ay2 - Ay1);
        area += (Bx2 - Bx1) * (By2 - By1);
        int w;
        int h;
        w = Math.max(0, Math.min(Ax2, Cx2) - Math.max(Ax1, Cx1));
        h = Math.max(0, Math.min(Ay2, Cy2) - Math.max(Ay1, Cy1));
        area -= w * h;
        w = Math.max(0, Math.min(Bx2, Cx2) - Math.max(Bx1, Cx1));
        h = Math.max(0, Math.min(By2, Cy2) - Math.max(By1, Cy1));
        area -= w * h;
        System.out.printf("  :| area=%s\n", area);
        out = new PrintWriter(new File("billboard.out"));
        out.printf("%d\n", area);
        out.close();
    }

}

File: 1801/harry/billboard.cpp
//Harry Zhou
//55 min
//used VIM this time
#include <cstdio>
#include <algorithm>
int main() {
    FILE *fileIn = fopen("billboard.in", "r");
    
    int Ax1, Ay1, Ax2, Ay2;
    int Bx1, By1, Bx2, By2;
    fscanf(fileIn, "%d %d %d %d", &Ax1, &Ay1, &Ax2, &Ay2);
    fscanf(fileIn, "%d %d %d %d", &Bx1, &By1, &Bx2, &By2);

    int blockX = 0;
    int blockY = 0;
    int area;
    if (Ax1 >= Bx1 && Ax2 <= Bx2) {
        blockX = 1;
    }
    if (Ay1 >= By1 && Ay2 <= By2) {
        blockY = 1;
    }

    if (blockX && blockY) {
        area = 0;
    }
    else if (blockX) {
        int yOb = std::max( 0 , std::min(Ay2-By1,By2-Ay1) );
        if (By1>Ay1 && By2 < Ay2) {
            yOb = 0;
        }
        area = (Ax2-Ax1) * (Ay2-Ay1 - yOb);
    }
    else if (blockY) {
        int xOb = std::max( 0 , std::min(Ax2-Bx1,Bx2-Ax1) );
        if (Bx1>Ax1 && Bx2 < Ax2) {
            xOb = 0;
        }
        area = (Ax2-Ax1 - xOb) * (Ay2-Ay1);
    }
    else {
        area = (Ax2-Ax1) * (Ay2-Ay1);
    }
    printf("%d\n",area);

    FILE *fileOut = fopen("billboard.out", "w");
    fprintf(fileOut, "%d\n", area);
}
